src/proc_loadavg.c

```diff
@@ -138,7 +138,10 @@ static void insert_node(struct load_node **n, int locate)
 {
 	struct load_node *f;
 
-	pthread_mutex_lock(&load_hash[locate].lock);
+
+	if(mutex_lock_interruptible(&load_hash[locate].lock)){
+		return NULL;
```

Comment:
this function has a `void` return type. Have you tried to compile and test this code?